### PR TITLE
Cardinality check for metrics

### PR DESCRIPTION
**Pull request by rexagod**
Allow reduced cardinality for metrics, so that the overall influx of
data reaching data-hub's telemeter is minimal.
* introduces new wrapper metrics to avoid collision
* disallows parsing original noobaa rules and uses custom rules instead,
  to avoid conflict with products dependent on noobaa metrics, ODF, for
  instance.

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

### CODE REVIEW

**Comment by rexagod** [comment]
/hold
Need to run this by @cloudbehl first.

File: controllers/noobaa-rules.yaml
Comment (jmolmo) [inline controllers/noobaa-rules.yaml:202]: Maybe we do not need this alert..Are we sure that the capacity can be used completely, I mean, probably we are going to have always some bytes "free". In any case it does not hurt ... this is a simple comment.

File: controllers/monitoring_alert.go
Comment (jmolmo) [inline controllers/monitoring_alert.go:220]: IMHO, a single quote will make more easy to understand the message. ---> "removing 'app' label", instead "removing app label"

File: controllers/monitoring_alert.go
Comment (jmolmo) [inline controllers/monitoring_alert.go:216]: It seems that this is just a "cleaning" task to revert a label set in previous versions. If i am not wrong, maybe a brief comment explaining "why" we do this will make the code more easy to understand.

File: controllers/noobaa-rules.yaml
Comment (rexagod) [inline controllers/noobaa-rules.yaml:33]: Made changes based on @jmolmo's review and a conversation with @cloudbehl on Friday. I've commented this out right now as I need to test these two metrics on a cluster before I incorporate them.

Comment (rexagod) [reply controllers/noobaa-rules.yaml:33]: @cloudbehl For these two metrics, I'm not getting anything in Prometheus.

<details>
<summary>Screenshots</summary>

Screenshot from 2022-07-13 13-29-28

image

</details>

Comment (cloudbehl) [reply controllers/noobaa-rules.yaml:33]: did you create the namespace bucket  and bucket class to see the availability of the metrics?

File: controllers/noobaa-rules.yaml
Comment (cloudbehl) [inline controllers/noobaa-rules.yaml:53]: Is it related to namespace buckets?

File: controllers/noobaa-rules.yaml
Comment (cloudbehl) [inline controllers/noobaa-rules.yaml:67]: Why do we need this in DF?

File: controllers/noobaa-rules.yaml
Comment (cloudbehl) [inline controllers/noobaa-rules.yaml:103]: same

File: controllers/noobaa-rules.yaml
Comment (cloudbehl) [inline controllers/noobaa-rules.yaml:128]: which buckets are these, is it backing store or NooBaa internal buckets?

File: controllers/noobaa-rules.yaml
Comment (rexagod) [reply controllers/noobaa-rules.yaml:2]: That's outdated. Please refer [this](https://docs.google.com/document/d/19JST--IQyxTa_330D7mjJZUTEi2dqJ9NXcJ9J0SrhyY/edit#).

File: controllers/noobaa-rules.yaml
Comment (rexagod) [reply controllers/noobaa-rules.yaml:33]: Yep, I created these and now they are available (as seen in the PDF I sent you). I've added them.

**Comment by rexagod** [comment]
@cloudbehl

- Removed all `.*Quota.*` alerts.
- Removed all `.*Capacity.*` alerts.
- Removed IOPS and throughput metrics.
- Added metrics for buckets and nss.

Please review.

File: controllers/noobaa-rules.yaml
Comment (cloudbehl) [inline controllers/noobaa-rules.yaml:81]: Instead of calling it namespace resource and bucket can we change the naming as per the product?

Bucket Class = Buckey Policy

Namespace Store = Data resource

File: controllers/noobaa-rules.yaml
Comment (cloudbehl) [inline controllers/noobaa-rules.yaml:44]: Can we name it like this?
```suggestion
       - alert:BucketPolicyErrorState
```

File: controllers/noobaa-rules.yaml
Comment (cloudbehl) [inline controllers/noobaa-rules.yaml:47]:
```suggestion
           description: A data source {{ $labels.bucket_name }} is in error
             state for more than 5m
```

File: controllers/noobaa-rules.yaml
Comment (cloudbehl) [inline controllers/noobaa-rules.yaml:48]:
```suggestion
           message: A Data Source Is In Error State
```

**Review by cloudbehl — COMMENTED** [review]
All metrics looks good.



Can we make changes to the alert as per the suggestion I provided for the first alert?

**Comment by rexagod** [comment]
Context for the change introduced in the last commit: https://github.com/red-hat-storage/mcg-osd-deployer/pull/61/commits/7aa6b963eb9ec6ad13be73d0f2d3cc1290d7e4f5

Screenshot from 2022-07-17 20-22-01

File: controllers/noobaa-rules.yaml
Comment (rexagod) [inline controllers/noobaa-rules.yaml:80]:
```suggestion
       - alert: CacheBucketErrorState
```

**Comment by rexagod** [comment]
/cc @cloudbehl @SanjalKatiyar @jmolmo 
PTAL and approve if LGTY, thanks.

**Comment by SanjalKatiyar** [comment]
LGTM

File: controllers/noobaa-rules.yaml
Comment (cloudbehl) [inline controllers/noobaa-rules.yaml:105]: I think we should fix the Namespace store usage to Data resource as well?

**Review by cloudbehl — CHANGES_REQUESTED** [review]

**Comment by rexagod** [comment]
/cc @naveenpaul1
PTAL and LGTM if this looks fine to you, thanks.

File: controllers/monitoring_alert.go
Comment (naveenpaul1) [inline controllers/monitoring_alert.go:221]: one doubt, we are removing the original Prometheus rule to add the labels origin: mcg-osd-deployer which will help us to identify our metrics? if yes where are we adding this labels to metrics? @rexagod

Comment (rexagod) [reply controllers/monitoring_alert.go:221]: We are removing the original rule file to remove/add/edit a bunch of custom rules and alerts, in order to provide clarity to the user. We are adding these labels here: https://github.com/red-hat-storage/mcg-osd-deployer/pull/61/files#diff-c2d8ea863a8b41a65ff336525a37d8a9b198d683b1dae92c7d4be6f06c446d7eR7.

File: controllers/noobaa-rules.yaml
Comment (simonpasquier) [inline controllers/noobaa-rules.yaml:69]: This should be `summary`.

File: controllers/noobaa-rules.yaml
Comment (simonpasquier) [inline controllers/noobaa-rules.yaml:71]: this looks more like a label though this could also be conveyed in the alert name (e.g. `BucketPolicyErrorState` -> `NooBaaBucketPolicyErrorState`)

**Pull request merged**
